libraries/persistent_store/src/format.rs

```diff
@@ -0,0 +1,901 @@
+        }
+        Ok(WordState::Valid(valid))
+    }
+
+    /// Builds the storage representation of a user entry.
+    pub fn build_user(&self, key: usize, value: &[u8]) -> Vec<u8> {
+        let length = value.len();
+        let word_size = self.word_size();
+        let footer = self.bytes_to_words(length);
+        let mut result = vec![0xff; (1 + footer) * word_size];
+        result[word_size..][..length].copy_from_slice(value);
+        let mut word = ERASED_WORD;
+        ID_HEADER.set(&mut word);
+        if footer > 0 && is_erased(&result[footer * word_size..]) {
+            HEADER_FLIPPED.set(&mut word);
+            *result.last_mut().unwrap() = 0x7f;
+        }
+        HEADER_LENGTH.set(&mut word, length);
+        HEADER_KEY.set(&mut word, key);
+        HEADER_CHECKSUM.set(&mut word, count_zeros(&result[footer * word_size..]));
+        result[..word_size].copy_from_slice(&word.to_ne_bytes());
+        result
+    }
+
+    /// Sets the padding bit in the first word of a user entry.
+    pub fn set_padding(&self, word: &mut WORD) {
+        ID_PADDING.set(word);
+    }
+
+    /// Sets the deleted bit in the first word of a user entry.
+    pub fn set_deleted(&self, word: &mut WORD) {
+        HEADER_DELETED.set(word);
+    }
+
+    /// Returns the minimum number of words to represent a given number of bytes.
+    ///
+    /// Assumes `bytes + self.word_size()` does not overflow.
+    pub fn bytes_to_words(&self, bytes: usize) -> usize {
+        div_ceil(bytes, self.word_size())
+    }
+}
+
+/// The word index of the init info in a page.
+const INIT_WORD: usize = 0;
+
+/// The word index of the compact info in a page.
+const COMPACT_WORD: usize = 1;
+
+/// The word index of the content of a page.
+///
+/// Since a page is at least 8 words, there is always at least 6 words of content.
+const CONTENT_WORD: usize = 2;
+
+/// The checksum for a single word.
+///
+/// Since checksums are the number of bits set to zero and a word is 32 bits, we need 5 bits to
+/// store numbers between 0 and 27 (which is 32 - 5).
+const WORD_CHECKSUM: Checksum = Checksum {
+    field: Field { pos: 27, len: 5 },
+};
+
+// The fields of the init info of a page.
+bitfield! {
+    /// The number of times the page has been erased.
+    INIT_CYCLE: Field <= MAX_ERASE_CYCLE,
+
+    /// The word index of the first entry in this virtual page.
+    INIT_PREFIX: Field <= div_ceil(MAX_VALUE_LEN, WORD_SIZE),
+
+    #[cfg(test)]
+    LEN_INIT: Length,
+}
+
+// The fields of the compact info of a page.
+bitfield! {
+    /// The distance in words between head and tail at compaction.
+    ///
+    /// In particular, compaction copies non-deleted user entries from the head to the tail as long
+    /// as entries span the page to be compacted.
+    COMPACT_TAIL: Field <= MAX_VIRT_PAGE_SIZE * MAX_PAGE_INDEX,
+
+    #[cfg(test)]
+    LEN_COMPACT: Length,
+}
+
+// Overview of the first word of the different kind of entries:
+//
+//         0123456789abcdef0123456789abcdef
```

Comment:
What's this hex value's meaning?

Reply:
I clarified in the comment.

Reply:
I don't see it done in the latest version of this file.

Reply:
You probably need to look at the latest version of the file.